app/src/main/java/org/kinecosystem/kinit/repository/CategoriesRepository.kt

```diff
@@ -0,0 +1,182 @@
+        this.headerMessage = headerMsg
+        headerMessage?.let {
+            headerTitle.set(it.title)
+            headerSubtitle.set(it.subtitle)
+        }
+        categoriesCache.getString(HEADER_MESSAGE_KEY, Gson().toJson(headerMsg))
+    }
+
+    private fun updateFromCache() {
+        updateCategories(Gson().fromJson(categoriesCache.getString(CATEGORIES_KEY, ""), object : TypeToken<List<Category>>() {}.type))
+        updateHeader(Gson().fromJson(categoriesCache.getString(HEADER_MESSAGE_KEY, ""), HeaderMessage::class.java))
+    }
+
+    fun updateTasks(updatedTasks: Map<String, List<Task>>) {
+        Log.d("###", "#### updateTasks updateTasks ")
+
+        //TOOD check if need to replace!!!!
+        for ((catId, tasksList) in updatedTasks) {
+            if (tasks[catId] == null) {
+                Log.d("###", "#### updateTasks null new repo $catId")
+                tasks[catId] = TasksRepo(catId, tasksList.firstOrNull())
+            } else {
+                Log.d("###", "#### update tasksk   $catId  ${tasks[catId]}")
+                tasks[catId]?.task = tasksList.firstOrNull()
+                // val repos = tasks[catId]
+            }
+        }
+    }
+
+    private fun getTaskRepo(): TasksRepo? {
+        currentCategoryId?.let {
+            return if (tasks.containsKey(it)) {
+                tasks.getValue(it)
+            } else null
+        }
+        return null
+    }
+
+    fun getCurrentTaskState() = currentTaskRepo?.taskState
+
+    fun isTaskComplete(): Boolean {
+        currentTaskRepo?.let {
+            return it.isTaskComplete()
+        }
+        return false
+    }
+
+    fun getTask(categoryId: String): Task? {
+        return if (tasks.containsKey(categoryId)) {
+            return tasks.getValue(categoryId)?.task
+        } else null
+    }
+
+    fun getCategory(categoryId: String): Category? {
+        for (category in categories.get()) {
+            if (categoryId.equals(category.id)) {
+                return category
+            }
+        }
+        return null
+    }
+
+    fun updateAvailableTasksCount(categoryId: String, taskCount: Int) {
+        currentAvailableTasks.set(taskCount)
+        for (category in categories.get()) {
+            if (categoryId == category.id) {
+                category.availableTasksCount = taskCount
+            }
+        }
+    }
+
+    fun updateTaskState(taskId: String, newState: Int) {
+        if (taskId == currentTaskRepo?.taskInProgress?.id) {
+            currentTaskRepo?.taskState = newState
+        }
+    }
+
+    fun onCategorySelected(category: Category? = null) {
+        currentCategoryId = category?.id
+        currentTaskRepo = getTaskRepo()
+        category?.availableTasksCount?.let { currentAvailableTasks.set(it) }
+        Log.d("#####", "#### cat id $currentCategoryId  task repo $currentTaskRepo task in progress $currentTaskInProgress")
+    }
+
+    fun getNumOfAnsweredQuestions(): Int {
```

Comment:
It doesn't sound right to expose this metho at the level of the category Repo.
The method name does not say for which task we are retrieving the num of answered questions.
Would rather have other objects do something like:
'categoryRepo.currentTaskInProgress.numOfAnsweredQuestions'